nengo_gui/components/component.py

```diff
@@ -82,13 +82,16 @@ def remove_nengo_objects(self, page):
         """
         pass
 
-    def javascript_config(self, cfg):
+    def javascript_config(self, cfg, override = {}):
         """Convert the nengo.Config information into javascript.
         
         This is needed so we can send that config information to the client.
         """
         for attr in self.config._clsparams.params:
-            cfg[attr] = getattr(self.config, attr)
+            if attr not in override:
+                cfg[attr] = getattr(self.config, attr)
+            else:
+                cfg[attr] = override[attr]
```

Comment:
See the comment in bgplot.py; I'm pretty sure that this override thing isn't needed at all, since you can just directly put that stuff in the dictionary you're creating and passing in here.